lib/features/thread/data/extensions/list_email_extension.dart

```diff
@@ -17,23 +17,17 @@ extension ListEmailExtension on List<Email> {
     };
   }
 
-  List<Email> sortingByOrderOfIdList(List<Id> ids) {
-    if (ids.length != length) {
-      return this;
-    }
+  List<Email> sortEmailsById(List<Id> referenceIds) {
+    final indexMap = {
+      for (var i = 0; i < referenceIds.length; i++)
+        referenceIds[i]: i
+    };
 
     sort((email1, email2) {
-      final id1 = email1.id?.id;
-      final id2 = email2.id?.id;
-
-      if (id1 == null || id2 == null) {
-        return 0;
-      }
-
-      final index1 = ids.indexWhere((id) => id == id1);
-      final index2 = ids.indexWhere((id) => id == id2);
+      final indexEmail1 = indexMap[email1.id!.id] ?? double.maxFinite;
+      final indexEmail2 = indexMap[email2.id!.id] ?? double.maxFinite;
```

Comment:
`!` is not safe here. How about `?`?

Reply:
It is safe, as per jmap documentation. Email always has id, so it is always non-null.

Reply:
But in client side it is nullable, it also depends in get properties, so it should be safer in our side

Reply:
So we should change them all to a non-null property. That's the right thing to do. We shouldn't care too much about the cases when it's optional.

Reply:
updated